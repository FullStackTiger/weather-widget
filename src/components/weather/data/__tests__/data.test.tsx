import { render } from "@testing-library/react";
import { WeatherData, useWeatherData } from "../";
import { useApi } from "../../../../api/use-api";

jest.mock("../../../../api/use-api", () => ({
  useApi: jest.fn(),
}));

beforeEach(() => jest.clearAllMocks());

describe("useWeatherData", () => {
  it("should call Weather Data endpoint", () => {
    useWeatherData();
    expect(useApi).toHaveBeenCalledTimes(1);
  });
});

describe("WeatherData", () => {
  it("should return false", () => {
    (useApi as jest.Mock).mockReturnValue({
      isLoading: true,
    });

    const { baseElement } = render(<WeatherData />);
    expect(baseElement).toMatchSnapshot();
  });

  it("should return error", () => {
    (useApi as jest.Mock).mockReturnValue({
      isLoading: false,
      error: new Error('foobar'),
    });
    const { baseElement } = render(<WeatherData />);
    expect(baseElement).toMatchSnapshot();
  });
});
