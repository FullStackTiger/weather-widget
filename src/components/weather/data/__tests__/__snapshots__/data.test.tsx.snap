// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherData should return error 1`] = `
<body>
  <div>
    <div
      class="alert-container error"
    >
      foobar
    </div>
  </div>
</body>
`;

exports[`WeatherData should return false 1`] = `
<body>
  <div>
    <div
      class="spinner-container"
    >
      <div
        class="spinner"
      />
    </div>
  </div>
</body>
`;
